tests/test_designmatrix_noisepars.py

```diff
@@ -0,0 +1,46 @@
+import os
+import pint.models.model_builder as mb
+import pinttestdata
+import pytest
+
+
+@pytest.fixture()
+def model_and_toas():
+    datadir = pinttestdata.datadir
+    parfile = os.path.join(datadir, "J1713+0747_NANOGrav_11yv0_short.gls.par")
+    timfile = os.path.join(datadir, "J1713+0747_NANOGrav_11yv0_short.tim")
+    return mb.get_model_and_toas(parfile, timfile)
+
+
+def _test_designmatrix(model, toas):
+    try:
+        # This should raise a NotImplementedError rather than an AttributeError.
+        M = model.designmatrix(toas)
+    except NotImplementedError:
+        pass
+
+    # This should work.
+    M = model.designmatrix(toas, incnoise=False)
+
+
+def test_designmatrix_noise_exception_falsepositive(model_and_toas):
```

Comment:
This title is unclear. Perhaps in this situation no noise parameter is free and you want to make sure the design matrix can be created? Can the title say that?

Reply:
I have changed the function name so that it is clearer.